tools/bridge/bridge/transfer_recorder.py

```diff
@@ -49,6 +58,25 @@ def apply_proper_event(self, event: AttributeDict) -> None:
         else:
             raise ValueError(f"Got unknown event {event}")
 
+    def apply_control_event(self, event: ControlEvent) -> None:
+        if isinstance(event, IsValidatorCheck):
+            if event.is_validator and not self.is_validator:
+                logger.info("Account is a member of the validator set")
+                self.is_validator = True
+            elif not event.is_validator and self.is_validator:
+                logger.info("Account is not a member of the validator set")
+                self.is_validator = False
+
+        elif isinstance(event, BalanceCheck):
+            balance_sufficient_before = self.is_balance_sufficient
+            self.balance = event.balance
+            balance_sufficient_now = self.is_balance_sufficient
+
+            if balance_sufficient_now and not balance_sufficient_before:
+                logger.info("Account balance is sufficiently high")
+            elif not balance_sufficient_now and balance_sufficient_before:
+                logger.info("Account balance is below safe minimum")
```

Comment:
Please log that with logger.warn and with the balance. Always log it when a new event comes in to make it clear something is wrong.